ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/ConnectionProxy.java

```diff
@@ -72,6 +72,16 @@ public Session createSession(boolean transacted, int acknowledgeMode) throws JMS
         return xaConnection.createSession(transacted, acknowledgeMode);
     }
 
+    @Override
+    public Session createSession(int sessionMode) throws JMSException {
+        return xaConnection.createSession(sessionMode);
```

Comment:
It should check the JTA transaction is active and register the XAResource just like the above method
```
createSession(boolean trasacted, int acknowledgeMode);
```